src/main.rs

```diff
@@ -0,0 +1,83 @@
+use mdbook::renderer::RenderContext;
+use serde::Deserialize;
+use std::collections::BTreeMap;
+use std::io;
+
+#[derive(Deserialize)]
+struct I18nConfiguration {
+    languages: BTreeMap<String, String>,
+    default_language: Option<String>,
+
+    #[serde(default)]
+    translate_all_languages: bool,
+    #[serde(default)]
+    move_translations_to_html_directory: bool,
+}
+
+fn main() {
+    let mut stdin = io::stdin();
+
+    // Get the configs
+    let ctx = RenderContext::from_json(&mut stdin).unwrap();
+    let i18n_config: I18nConfiguration = ctx
+        .config
+        .get_deserialized_opt("output.i18n-helpers")
+        .unwrap()
+        .unwrap();
+
+    if !i18n_config.translate_all_languages {
+        return;
+    }
+
+    let mut mdbook = mdbook::MDBook::load(&ctx.root).expect("Failed to load book");
+    // Overwrite with current values from stdin.
+    mdbook.book = ctx.book.clone();
+    mdbook.config = ctx.config.clone();
+    mdbook.root = ctx.root.clone();
+
+    let book_config = mdbook
+        .config
+        .get_mut("output.i18n-helpers")
+        .expect("No output.i18n-helpers config in book.toml");
+    // Set translate_all_languages to false for nested builds to prevent infinite recursion.
```

Comment:
I wonder what the use case if for having `translate_all_languages`? Could we not decide that it is a configuration mistake to enable the `i18n` backend if you don't want to translate all languages? So just write the code as if it was always true.

Reply:
For performance, if you have many translations, and a potato computer, it could be slow